src/OpenTelemetry.Extensions/Trace/PartialActivityProcessor.cs

```diff
@@ -0,0 +1,273 @@
+        logRecord.Body = Convert.ToBase64String(buffer, 0, writePosition);
+
+        // Severity = LogRecordSeverity.Info,
+        // SeverityText = "Info",
+
+        var logRecordAttributes = GetLogRecordAttributes();
+        logRecordAttributes.AddRange(logRecordAttributesToBeAdded);
+        logRecord.Attributes = logRecordAttributes;
+
+        return logRecord;
+    }
+
+    private static void AccessOpenTelemetryInternals(
+        out MethodInfo writeTraceDataMethodParam,
+        out ConstructorInfo logRecordConstructorParam,
+        out object sdkLimitOptionsParam)
+    {
+        var sdkLimitOptionsType = Type.GetType(
+            "OpenTelemetry.Exporter.OpenTelemetryProtocol.Implementation.SdkLimitOptions, OpenTelemetry.Exporter.OpenTelemetryProtocol",
+            true);
+
+        if (sdkLimitOptionsType == null)
+        {
+            throw new InvalidOperationException("Failed to get the type 'SdkLimitOptions'.");
+        }
+
+        sdkLimitOptionsParam = Activator.CreateInstance(sdkLimitOptionsType, nonPublic: true) ??
+                          throw new InvalidOperationException(
+                              "Failed to create an instance of 'SdkLimitOptions'.");
+
+        var protobufOtlpTraceSerializerType = Type.GetType(
+            "OpenTelemetry.Exporter.OpenTelemetryProtocol.Implementation.Serializer.ProtobufOtlpTraceSerializer, OpenTelemetry.Exporter.OpenTelemetryProtocol",
+            true);
+
+        if (protobufOtlpTraceSerializerType == null)
+        {
+            throw new InvalidOperationException(
+                "Failed to get the type 'ProtobufOtlpTraceSerializer'.");
+        }
+
+        writeTraceDataMethodParam =
+            protobufOtlpTraceSerializerType.GetMethod(
+                "WriteTraceData",
+                BindingFlags.NonPublic | BindingFlags.Static) ??
+            throw new InvalidOperationException("Failed to get the method 'WriteTraceData'.");
+
+        var logRecordType = Type.GetType("OpenTelemetry.Logs.LogRecord, OpenTelemetry", true);
+
+        if (logRecordType == null)
+        {
+            throw new InvalidOperationException("Failed to get the type 'LogRecord'.");
+        }
+
+        logRecordConstructorParam = logRecordType.GetConstructor(
+                                   BindingFlags.NonPublic | BindingFlags.Instance,
+                                   null,
+                                   Type.EmptyTypes,
+                                   null) ??
+                               throw new InvalidOperationException(
+                                   "Failed to get the constructor of 'LogRecord'.");
+    }
+
+    private static List<KeyValuePair<string, object?>> GetLogRecordAttributes() =>
+    [
+        new("telemetry.logs.cluster", "partial"),
+        new("telemetry.logs.project", "span"),
+    ];
+
+    private static List<KeyValuePair<string, object?>> GetStopLogRecordAttributes() =>
+    [
+        new("partial.event", "stop"),
```

Comment:
maybe

```suggestion
        new("span.state", "ended"),
```

We usually say that span has "started" and "ended".
I know that it is not great, but I think that ship has sailed: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#interface-definition.

Reply:
Sure. This was more of internal idea of naming convention.